File: patch_rules.mjs
import fs from 'fs';
const fp = 'firestore.rules';
let code = fs.readFileSync(fp, 'utf8');

const newRule = `    // Reports (Anti-Cheat)
    match /reports/{reportId} {
      allow create: if isSignedIn() && incoming().reporterId == request.auth.uid;
      allow read: if false; // Only admins/backend can read
    }
  }
}`;

code = code.replace("  }\n}", newRule);
fs.writeFileSync(fp, code);
console.log("Patched rules");

File: patch_rules.py
with open("firestore.rules", "r") as f:
    content = f.read()

content = content.replace("      // Only players in the game can update it\n      allow update: if isSignedIn()\n        && (existing().whiteId == request.auth.uid || existing().blackId == request.auth.uid);", 
"      // Only players in the game can update it, OR a user joining an invite\n      allow update: if isSignedIn()\n        && (\n          existing().whiteId == request.auth.uid || \n          existing().blackId == request.auth.uid ||\n          (existing().status == 'waiting_friend' && existing().blackId == '' && incoming().blackId == request.auth.uid)\n        );")

with open("firestore.rules", "w") as f:
    f.write(content)
